lib/net.js

```diff
@@ -147,6 +156,7 @@ function initSocketHandle(self) {
   if (self._handle) {
     self._handle.owner = self;
     self._handle.onread = onread;
+    self[async_id_symbol] = getNewAsyncId(self._handle);
```

Comment:
Why is the `getAsyncId` method not guaranteed to exists? Even if it doesn't exists, is `newUid` really sufficient and do we risk overwriting `async_id_symbol`?